### PR TITLE
[SPARK-24304][SS] Scheduler changes for continuous processing shuffle support

### PR DESCRIPTION
**Pull request by xuanyuanking**
## What changes were proposed in this pull request?

This is the last part of the preview PRs, the mainly change is in DAGScheduler, we added a interface to submit whole job at once.
In order to work and compile properly, this also including several parts in #21332 and #21293. I will marked the duplicated part for convenient review. Sorry about this...

## How was this patch tested?

Added new UT

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/Dependency.scala [hunk 1]
```diff
@@ -88,14 +96,53 @@ class ShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
...
 }
 
+/**
+ * :: DeveloperApi ::
+ * Represents a dependency on the output of a shuffle stage of continuous type.
+ * Different with ShuffleDependency, the continuous dependency only create on Executor side,
+ * so the rdd in param is deserialized from taskBinary.
+ */
+@DeveloperApi
+class ContinuousShuffleDependency[K: ClassTag, V: ClassTag, C: ClassTag](
```
Comment (xuanyuanking) [inline core/src/main/scala/org/apache/spark/Dependency.scala]: Changes about ShuffleDependency same in #21293

File: core/src/main/scala/org/apache/spark/MapOutputTracker.scala [hunk 2]
```diff
@@ -213,6 +213,12 @@ private[spark] sealed trait MapOutputTrackerMessage
 private[spark] case class GetMapOutputStatuses(shuffleId: Int)
   extends MapOutputTrackerMessage
 private[spark] case object StopMapOutputTracker extends MapOutputTrackerMessage
+private[spark] case class CheckNoMissingPartitions(shuffleId: Int)
```
Comment (xuanyuanking) [inline core/src/main/scala/org/apache/spark/MapOutputTracker.scala]: Changes about MapOutputTracker same in #21293

File: core/src/main/scala/org/apache/spark/SparkEnv.scala [hunk 3]
```diff
@@ -140,6 +140,7 @@ object SparkEnv extends Logging {
 
   private[spark] val driverSystemName = "sparkDriver"
   private[spark] val executorSystemName = "sparkExecutor"
+  private[spark] val START_EPOCH_KEY = "__continuous_start_epoch"
```
Comment (xuanyuanking) [inline core/src/main/scala/org/apache/spark/SparkEnv.scala]: Changes about SparkEnv same in #21293

**Comment by jose-torres** [comment]
As I've mentioned elsewhere, stages are currently submitted sequentially. That is, for a stage X, all the stage dependencies of X are completed before the tasks within X start. This change proposes to violate that invariant, and it's not obvious that this is a safe approach. The questions we need to answer are:
* How can we attempt to validate that this is indeed safe to change, and will not break the scheduler or things dependent on it in subtle ways?
* What benefits do we derive from adding the additional risk of a scheduler change, rather than handling continuous shuffles entirely at the RDD layer?

**Comment by gaborgsomogyi** [comment]
@xuanyuanking are you still working on this? If not may I ask to close?

**Comment by xuanyuanking** [comment]
Yeah, I'll close all the outdated preview PR.

**Pull request closed without merging**